code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1027,10 +1115,20 @@ func (sc *StorageSmartContract) updateAllocationRequestInternal(
 			err.Error())
 	}
 
+	isEnterprise := false
+	_ = chainstate.WithActivation(balances, "electra", func() error {
+		return nil
+	}, func() error {
+		if v2 := sa.Entity().(*storageAllocationV2); v2.IsEnterprise != nil && *v2.IsEnterprise {
```

Comment:
Perhaps check if the type assertion is success, otherwise if it's v1, the `v2.IsEnterprise` could panic due to nil v2. Or check the entity.GetVersion() to see if it's v2.

Reply:
Added the check if v2 == nil.